ngraph/core/src/op/shape_of.cpp

```diff
@@ -152,6 +153,73 @@ namespace shape_of
         }
         return false;
     }
+
+    bool evaluate_bound_shape(const Node* shape_of_node,
+                              const HostTensorVector& output_values,
+                              bool is_upper)
+    {
+        const auto& input_partial_shape = shape_of_node->get_input_partial_shape(0);
+        if (input_partial_shape.rank().is_dynamic())
+            return false;
+        const auto rank = input_partial_shape.rank().get_length();
+        auto pshape_low = PartialShape::dynamic(rank), pshape_up = PartialShape::dynamic(rank);
+        for (Dimension::value_type i = 0; i < rank; ++i)
+        {
+            Interval interval = input_partial_shape[i].get_interval();
+            pshape_low[i] = interval.get_min_val();
+            pshape_up[i] = Dimension(interval.get_max_val()).is_dynamic()
+                               ? Dimension(interval.get_max_val() - 1)
+                               : interval.get_max_val();
+        }
+        NGRAPH_CHECK(pshape_up.is_static() && pshape_low.is_static());
+        const auto input_et = shape_of_node->get_input_element_type(0);
+        const auto output_et = shape_of_node->get_output_element_type(0);
+        if (pshape_low.to_shape() == pshape_up.to_shape())
+        {
+            shape_of_node->evaluate(output_values,
+                                    {std::make_shared<HostTensor>(input_et, pshape_low)});
+            shape_of_node->get_output_tensor(0).set_lower_value(output_values[0]);
```

Comment:
You never check the size of `output_values` but `evaluate_lower` and `evaluate_upper` methods are public API. We need to check and control arguments for these methods.